### ISSUE TITLE
Simulation ends with "access violation"

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#403]
```
Application Version: 5.1.6 Parallel 64 bit
SVN Revision Number: 1717
Compile Date: 25 May 2008
Operating System: XP Pro x64

Describe details of the issue below:

My simulation ends at approximately the same time (about 60s) every time I 
try to run it. Below is the command prompt text:

C:\Sim\test3>mpiexec -n 6 fds5_mpi_64 test3.fds
Process   4 of   5 is running on AKPC275.finnmapcons.fi
Process   1 of   5 is running on AKPC275.finnmapcons.fi
Process   3 of   5 is running on AKPC275.finnmapcons.fi
Process   5 of   5 is running on AKPC275.finnmapcons.fi
Process   2 of   5 is running on AKPC275.finnmapcons.fi
Process   0 of   5 is running on AKPC275.finnmapcons.fi
Mesh   1 is assigned to Process   0
Mesh   2 is assigned to Process   1
Mesh   3 is assigned to Process   2
Mesh   4 is assigned to Process   3
Mesh   5 is assigned to Process   4
Mesh   6 is assigned to Process   5

 Fire Dynamics Simulator

 Compilation Date : Sun, 25 May 2008
 Version          : 5.1.6 Parallel
 SVN Revision No. : 1717

 Job TITLE        :
 Job ID string    : test3

 Time Step:      1,    Simulation Time:      0.14 s
 Time Step:      2,    Simulation Time:      0.29 s
 Time Step:      3,    Simulation Time:      0.43 s
 Time Step:      4,    Simulation Time:      0.58 s
 Time Step:      5,    Simulation Time:      0.72 s
 Time Step:      6,    Simulation Time:      0.87 s
 Time Step:      7,    Simulation Time:      1.01 s
 Time Step:      8,    Simulation Time:      1.15 s
 Time Step:      9,    Simulation Time:      1.30 s
 Time Step:     10,    Simulation Time:      1.44 s
 Time Step:     20,    Simulation Time:      2.89 s
 Time Step:     30,    Simulation Time:      4.33 s
 Time Step:     40,    Simulation Time:      5.77 s
 Time Step:     50,    Simulation Time:      7.22 s
 Time Step:     60,    Simulation Time:      8.66 s
 Time Step:     70,    Simulation Time:     10.10 s
 Time Step:     80,    Simulation Time:     11.28 s
 Time Step:     90,    Simulation Time:     12.41 s
 Time Step:    100,    Simulation Time:     13.53 s
 Time Step:    200,    Simulation Time:     21.54 s
 Time Step:    300,    Simulation Time:     28.01 s
 Time Step:    400,    Simulation Time:     32.81 s
 Time Step:    500,    Simulation Time:     36.14 s
 Time Step:    600,    Simulation Time:     39.46 s
 Time Step:    700,    Simulation Time:     42.60 s
 Time Step:    800,    Simulation Time:     45.56 s
 Time Step:    900,    Simulation Time:     48.39 s
 Time Step:   1000,    Simulation Time:     50.82 s
 Time Step:   1100,    Simulation Time:     53.40 s
 Time Step:   1200,    Simulation Time:     55.92 s
forrtl: severe (157): Program Exception - access violation
Image              PC                Routine            Line        Source

ntdll.dll          0000000077EF07E2  Unknown               Unknown  Unknown
ntdll.dll          0000000077EF5FD0  Unknown               Unknown  Unknown
fds5_mpi_64.exe    000000000080FEFC  Unknown               Unknown  Unknown
fds5_mpi_64.exe    000000000077AB4A  Unknown               Unknown  Unknown
fds5_mpi_64.exe    000000000046BE93  MEMORY_FUNCTIONS_         159  
func.f90
fds5_mpi_64.exe    0000000000766DF4  FDS._MESH_EXCHANG        1769  
main_mpi.f90

fds5_mpi_64.exe    00000000007533B1  MAIN__                    732  
main_mpi.f90

fds5_mpi_64.exe    0000000000834860  Unknown               Unknown  Unknown
fds5_mpi_64.exe    000000000080F5BF  Unknown               Unknown  Unknown
kernel32.dll       0000000077D5964C  Unknown               Unknown  Unknown

job aborted:
rank: node: exit code[: error message]
0: AKPC275.finnmapcons.fi: 123
1: AKPC275.finnmapcons.fi: 123
2: AKPC275.finnmapcons.fi: 123
3: AKPC275.finnmapcons.fi: 123
4: AKPC275.finnmapcons.fi: 123
5: AKPC275.finnmapcons.fi: 157: process 5 exited without calling finalize

C:\Sim\test3>



I have tried adding FLUSH_FILE_BUFFERS=.FALSE. but that didn't help. The 
smoke exhaust system is activated at 60s so I tried to change that time 
but that didn't help either. The restart files are created every 10s, but 
it works everytime before 60s, so I don't think it's the problem.

Attached is the input file, it's made with Pyrosim and I have only used it 
to try and find possible problems, so it's a bit unrealistic with a very 
fast and short fire duration (basically the time is divided by ten) etc.

Anybody have an idea of what the problem could be?

Thanks!

Dan Pada
```

Original issue reported on code.google.com by `dan.pada@ako.fi` on 2008-07-28 11:52:52

<hr>
- _Attachment: [test3.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-424/comment-0/test3.fds)_

**Comment by gforney** [comment]
```
I'll take a look at it.
```

Original issue reported on code.google.com by `mcgratta` on 2008-07-28 12:26:48
- **Labels added**: OpSys-Windows, Performance

**Comment by gforney** [comment]
```
The calculations uses many droplets, and the error is related to the re-allocation 
of memory for these droplets. However, I cannot reproduce the error on my linux 
machines, and I do not have a 64 bit XP machine. I am also running the latest 
version of FDS. We plan to release FDS 5.2 this week. When we do, try the case 
again, and if the problem remains, we will attempt to debug it again. It is possible

that the problem has already been fixed.
```

Original issue reported on code.google.com by `mcgratta` on 2008-07-28 14:10:01

**Comment by gforney** [comment]
```
Have you tried FDS 5.2, and if so, does it solve the problem?
```

Original issue reported on code.google.com by `mcgratta` on 2008-08-08 19:42:50

**Comment by gforney** [comment]
```
Assume Verified.
```

Original issue reported on code.google.com by `mcgratta` on 2008-08-22 21:54:19

**Comment by gforney** [comment]
```
Sorry for the delay in my answer, been on travel.
The problem seems to be solved with fds 5.2 yes, it runs without interruptions.
Thanks!
```

Original issue reported on code.google.com by `dan.pada@ako.fi` on 2008-08-28 05:30:49
